### PR TITLE
tighten Trademark restricts.  No more "for Mozilla|Firefox"

### LINKED ISSUE #9267: Don't allow use of "Mozilla" or "Firefox" in add-on names
Companion issue for https://github.com/mozilla/addons-linter/issues/2158

Up until now, we've allowed add-on names to end with "for Firefox" or "for Mozilla". For SEO reasons, we don't want to allow this anymore. XPIs with add-on names containing "Firefox" or "Mozilla" should not be allowed in the submission flow.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #9267

### CODE REVIEW

**Comment by wagnerand** [comment]
@jvillalobos Do we need to gate this behind a waffle in order to coordinate with updating the policies?

**Comment by eviljeff** [comment]
ugh, forgot about that (it's a requirement that's added as a note in the project, rather than in each issue)

File: src/olympia/addons/tests/test_utils_.py [hunk 1]
```diff
@@ -27,41 +29,70 @@
 
 
 @pytest.mark.django_db
-@pytest.mark.parametrize('name, allowed, email', (
+@pytest.mark.parametrize('name, allowed, email, content_optmzn_waffle', (
```
Comment (diox) [inline src/olympia/addons/tests/test_utils_.py]: `content_optmzn_waffle` is a weird name...

Comment (eviljeff) [reply src/olympia/addons/tests/test_utils_.py]: yeah... I was just trying to choose a name short enough to fit it within the line limit.  It'll be gone again in a few months after the changes launch.

**Pull request merged**